src/physics/utils/phys_comp.F90

```diff
@@ -245,60 +202,76 @@ subroutine phys_run2(dtime_phys, cam_runtime_opts, phys_state, phys_tend,  &
            read_initialized_variables=use_init_variables)
 
       ! Initialize the physics time step
-      call cam_ccpp_physics_timestep_initial(phys_suite_name, dtime_phys,     &
-           errmsg, errcode)
+      call cam_ccpp_physics_timestep_initial(phys_suite_name)
       if (errcode /= 0) then
          call endrun('cam_ccpp_physics_timestep_initial: '//trim(errmsg))
       end if
```

Comment:
The errcode is no longer being returned (should it be)?  The next section is checking the non-existent errcode coming back from cam_ccpp_physics_timestep_initial.  I thought errmsg and errcode were mandatory in every high level CCPP call into the framework.  Either errode needs to be returned, or the checking needs to be removed in all the places where this occurs.

Reply:
the error code is still being set (via a use statement in the cap and here) so it's potentially changing during cam_ccpp_physics_timestep_initial but is not being passed in. it's a bit... inscrutable

Reply:
@nusbaume - After discussing this with Courtney, I am worried that when this code uses multiple tasks, the tasks will be clobbering each other's errcode and errmsg.  The rule of thumb in CAM is that module level variables can be set in init phases, but run phases can not set module level variables.   This is because init is in single threaded code and run is split across MPI tasks.   But, I don't really know where CAM-SIMA is splitting across MPI tasks, as that was handled in CAM at the chunking level, which CAM-SIMA does not have.  So, I may be worried about this for nothing if the MPI level is higher in CAM-SIMA?